### PR TITLE
Add autoscaling policy

### LINKED ISSUE #99: Autoscaling Policy
When load is high, the worker group should scale up. When load is low it should scale down. I've written some configuration that uses cloudwatch alarms to trigger autoscaling policies based on Average CPU Utilization and Memory Utilization.
- Average CPU Utilization > 80% => Add node
- Average Memory Utilization > 80% => Add node
- Average CPU Utilization < 40% => Remove node
- Average Memory Utilization < 40% => Remove node

Cooldown 300 seconds.

The memory utilization metric requires running some code on the hosts, but I've integrated it with cloud-config.

Is there any interest in adding this functionality to Tack? If so I'm happy to submit a PR for discussion. There's definitely room for improvement in the policies themselves, but it's a good starting point.

### PR DESCRIPTION
**Pull request by adambom**
Fixes #99

### CODE REVIEW

File: modules/worker/cloud-config.tf [hunk 1]
```diff
@@ -216,6 +238,17 @@ write-files:
         copytruncate
       }
 
+  - path: /home/core/go-aws-mon.sh
+    permissions: '0755'
+    owner: core
+    content: |
+      #!/bin/bash -
+      if [ ! -x /home/core/go-aws-mon ]; then
+        wget -O /home/core/go-aws-mon https://github.com/a3linux/go-aws-mon/raw/aggregate/bin/go-aws-mon
```
Comment (danielwg) [inline modules/worker/cloud-config.tf]: @adambom  Shouldn't this be wget -O /home/core/go-aws-mon https://github.com/a3linux/go-aws-mon/raw/master/bin/go-aws-mon ? 

I get a 404 on https://github.com/a3linux/go-aws-mon/raw/aggregate/bin/go-aws-mon

Comment (adambom) [reply modules/worker/cloud-config.tf]: You're right - that's a typo - I will update

**Comment by yagonobre** [comment]
@adambom  https://github.com/kubernetes/kubernetes/issues/11935

**Comment by fuyufjh** [comment]
@adambom I am using this patch now. `go-aws-mon` seems not working out-of-box. How should I config it?

**Pull request closed without merging**